src/node_watchdog.cc

```diff
@@ -90,6 +90,30 @@ void Watchdog::Timer(uv_timer_t* timer) {
   uv_stop(&w->loop_);
 }
 
+void SigintWatchdog::Init(Environment* env, Local<v8::Object> target) {
+  Local<v8::FunctionTemplate> constructor = env->NewFunctionTemplate(New);
+  constructor->InstanceTemplate()->SetInternalFieldCount(1);
+  Local<v8::String> sigintWatchdogString =
+        FIXED_ONE_BYTE_STRING(env->isolate(), "SigintWatchdog");
+    constructor->SetClassName(sigintWatchdogString);
+  target->Set(env->context(), sigintWatchdogString,
+              constructor->GetFunction(env->context()).ToLocalChecked())
+              .Check();
+}
+
+void SigintWatchdog::New(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  auto isolate = args.GetIsolate();
+  auto env = node::Environment::GetCurrent(args);
+  auto self = args.This();
+  auto callback = args[0].As<v8::Function>();
+  auto watchdog = new SigintWatchdog(isolate, callback);
+  self->Set(env->context(), env->owner_symbol(),
+            v8::External::New(isolate, watchdog));
```

Comment:
Is this symbol property used somewhere? Where is the `new SigintWatchdog(...)` allocation released?

What you probably want to do is to create a subclass that inherits from `SigintWatchdog` and `HandleWrap` that deals with the additional features here, and has the `uv_async_t` as a member rather than a pointer to it. (You could make `HandleSigint()` virtual in that case and override it.)

(And, side note, I’d have a preference for not using `auto` everywhere because it makes the code easier to write but less readable/reviewable.)

Reply:
We don't have a strong policy on `auto` though, it's basically up to the PR author. As long as the RHS makes it clear enough what the type is, I don't have a problem with it.